src/System.Management.Automation/engine/Utils.cs

```diff
@@ -1547,6 +1544,21 @@ internal static bool IsComObject(object obj)
 
             return oldMode;
         }
+                
+        internal static string DisplayHumanReadableFileSize(long bytes)
+        {
+            return bytes switch
+            {
+                >= 1152921504606847000 => $"{(bytes / 1152921504606847000.0).ToString("0.000000000")} EB",
+                < 1152921504606847000 and >= 1125899906842624 => $"{(bytes / 1125899906842624.0).ToString("0.0000000")} PB",
+                < 1125899906842624 and >= 1099511627776 => $"{(bytes / 1099511627776.0).ToString("0.00000")} TB",
+                < 1099511627776 and >= 1073741824 => $"{(bytes / 1073741824.0).ToString("0.000")} GB",
+                < 1073741824 and >= 1048576 => $"{(bytes / 1048576.0).ToString("0.0")} MB",
+                < 1048576 and >= 1024 => $"{(bytes / 1024.0).ToString("0.0")} KB",
+                < 1024 and >= 0 => $"{bytes} Bytes",
```

Comment:
Given that we are more likely run into size that falls in `KB`, `MB`, or `GB` ranges, I suggest reversing the order to list branches from small range to large range.

Reply:
Done. Bear in mind though that the JIT is very intelligent in creating jumptables for switch statements and does its own sorting based on constant value, dividing into buckets, binary search, etc: https://github.com/dotnet/roslyn/blob/d4dab355b96955aca5b4b0ebf6282575fad78ba8/src/Compilers/Core/Portable/CodeGen/SwitchIntegralJumpTableEmitter.cs#L65-L94